Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -4893,6 +4894,7 @@ void WebPageProxy::commitProvisionalPage(IPC::Connection& connection, FrameIdent
         send(Messages::WebPage::Close());
 
     const auto oldWebPageID = m_webPageID;
+    provisionalPage = nullptr;
```

Comment:
Maybe we could:
1. Use `RefPtr provisionalPage = std::exchange(m_provisionalPage, nullptr);` at the beginning of this function
2. and use `provisionalPage.releaseNonNull()` below.

Reply:
If you insist on writing it this way, I think it will need a comment to explain why we're explicitly nulling out the data member here.